src/dataLoaders/components/collectorsWizard/verify/VerifyCollectorsStep.tsx

```diff
@@ -83,11 +83,17 @@ export class VerifyCollectorStep extends PureComponent<Props> {
     const {
       match: {
         params: {orgID},
+        url,
       },
+      onExit,
       onClearDataLoaders,
     } = this.props
     onClearDataLoaders()
-    this.props.history.push(`/orgs/${orgID}/load-data/telegrafs`)
+    if (url.includes('telegraf-plugins')) {
```

Comment:
new/existing telegraf can be configured from two separate places; from telegraf plugins page in `source` and telegraf page itself. This `if` statement checks where we are in the app and redirects user back to the telegraf page accordingly.